src/toxcore/dht/dht_node.rs

```diff
@@ -154,15 +154,31 @@ impl DhtNode {
                         None => Some(SocketAddr::V6(v6)),
                     }
                 },
-                None => {
-                    self.assoc4.saddr.map(Into::into)
-                },
+                None => self.assoc4.saddr.map(Into::into)
+                    .or_else(|| {
```

Comment:
We can just set variable `let result = ...` before returning the address. And then we can warn only if it's empty. This will reduce two checks/calls to one.

Reply:
Or we can check it right where we call this function